zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -3182,6 +3182,23 @@ public void addWatch(String basePath, Watcher watcher, AddWatchMode mode)
         }
     }
 
+    /**
+     * Add a watch to the given znode using the given mode. Note: not all
+     * watch types can be set with this method. Only the modes available
+     * in {@link AddWatchMode} can be set with this method. In this version of the method,
+     * the default watcher is used
+     *
+     * @param basePath the path that the watcher applies to
+     * @param mode type of watcher to add
+     * @throws InterruptedException If the server transaction is interrupted.
+     * @throws KeeperException If the server signals an error with a non-zero
+     *  error code.
+     */
+    public void addWatch(String basePath, AddWatchMode mode)
+            throws KeeperException, InterruptedException {
```

Comment:
The new added apis:`addWatch` needs  `@since 3.6.0` in its java doc?

Reply:
done